### ISSUE TITLE
Flags for PhySL interpreter do not output data to file arguments

### ISSUE THREAD

**Issue by kawilliams** [STEllAR-GROUP/phylanx#602]
I've been working on figuring out which flags to call with the physl interpreter but I'm having trouble for the options that require arguments (I want --dump-newick-tree and --dump-counters) . I've tried (I think) every combination of ```--dump-newick-tree "treefile.txt" ``` but I can't get my file to be created (with or without any of the correct output).

First I copied als.physl (from phylanx/examples/algorithms/als/) into phylanx/build/bin (the directory with ./physl). Then I ran:
```
srun -n 1 ./physl als.physl --performance
```
which prints the same contents that we usually see in the output file, except the trees and csv are printed to the command line (this is the output data that I want, except I want the Newick tree to automatically to go "tree.txt" and the performance counter csv to go to "performance.csv").

I can also do 
```
srun -n 1 ./physl als.physl --help
```

and see my options. But for an options like --dump-dot, I get:
```
[kwilliams@rostam1 bin]$ srun -n 1 ./physl als.physl --dump-dot "alsdot"
srun: job 3207598 queued and waiting for resources
srun: job 3207598 has been allocated resources
Linear Regression Algorithm example in PhySL
Usage: als.physl filepath row col regularization numfactors iterations alpha enableoutput
and no "alsdot"/alsdot.txt/alsdot/alsdot.dot file exists in the directory.
```

I've tried lots of permutations (e.g. srun -n 1 ./physl als.physl --dump-dot myfile; --dump-dot "myfile"; --dump-dot=myfile; --dump-dot="myfile") but none have written the output to myfile (or written any output, besides the "Linear Regression Algorithm example in PhySL ... enableoutput" lines).

For reference, here's what the -h flag shows for me to do:
Usage: physl <physl_script> [options] [arguments...]:

  -h [ --help ]                     print out program usage

  --dump-dot arg                    Write the topology of the created execution

**Comment by hkaiser** [comment]
@kawilliams the `als.physl` script expects additional arguments to be passed to it, i.e.
```
physl als.physl filepath row col regularization numfactors iterations alpha enableoutput
```
where:
- `filepath`: the path to the input file to use (usually MovieLens.csv)
- `row`, `col`: the number of rows/columns to use from the input file; `rows == 10`, `cols == 100` is fine, but any other - possibly smaller - values should work
- `regularization`: `0.1` should be fine
- `numfactors`: `10` is ok
- `iterations`: any number between `1` and the limit of your patience
- `alpha `: we normally use `40`
- `enableoutput`: `0` or `1`, depending on whether the script should print values after each iteration

Any additional option, like `--dump-dot` or similar can be appended to those options (or prepended or interspersed, it does not matter).

The bottom line is that the following should work:
```
srun -n1 ./physl als.physl MovieLens.csv 10 100 0.1 10 1 40 0 --dump-dot=als.dot
```
Could you please try again?

**Comment by hkaiser** [comment]
@kawilliams that is an unrelated error. Not sure what's wrong, but somehow your command line gets scrambled. Can you try running the same without srun? Do you see the same problem? Also, is the file MovieLens.csv in the current directory?

**Comment by kawilliams** [comment]
Unfortunately, I get the same error when I run without srun (yes, als.physl, physl, and MovieLens.csv are all in the current directory)
```
[kwilliams@rostam1 bin]$ ./physl als.physl MovieLens.csv 10 100 0.1 10 1 40 0 --dump-dot=2018-10-02-tree.dot
physl: exception caught:
Error! Incomplete parse: line 1:
MovieLens.csv
~~~~~~~~~^
: HPX(bad_parameter)
```

**Comment by scheherzade** [comment]
@kawilliams sorry I just saw this issue, I think your problem is related to how to pass a string to physl in command line, adding scape characters should solve your problem, just change MovieLens.csv to \"MovieLens.csv\"  as:
./physl als.physl \"MovieLens.csv\" 10 100 0.1 10 1 40 0 --dump-dot=2018-10-02-tree.dot

There is another issue with a missing parenthesis and also removed set_row primitive which I will create a pull request for very soon.

**Comment by kawilliams** [comment]
I still get the same error:
```
[kwilliams@rostam1 bin]$ ./physl als.physl "MovieLens.csv" 10 100 0.1 10 1 40 0 --dump-dot=2018-10-02-tree.dot
physl: exception caught:
Error! Incomplete parse: line 1:
MovieLens.csv
~~~~~~~~~^
: HPX(bad_parameter)
```

On a whim, I added quotes to als.physl but got the same error:
```
[kwilliams@rostam1 bin]$ ./physl "als.physl" "MovieLens.csv" 10 100 0.1 10 1 40 0 --dump-dot=2018-10-02-tree.dot
physl: exception caught:
Error! Incomplete parse: line 1:
MovieLens.csv
~~~~~~~~~^
: HPX(bad_parameter)
```

I also tried adding to the MovieLens.csv path:
```
[kwilliams@rostam1 bin]$ ./physl als.physl ./MovieLens.csv 10 100 0.1 10 1 40 0 --dump-dot=2018-10-02-tree.dot
physl: exception caught:
Error! Incomplete parse: line 1:
./MovieLens.csv
^
: HPX(bad_parameter)
```

**Comment by justwagle** [comment]
Try this:
./physlals.physl  \"/phylanx-data/CSV/MovieLens_20m.csv\"  10 100 0.1 10 1 40 0 --dump-dot=2018-10-02-tree.dot

**Comment by kawilliams** [comment]
Adding \" worked (thank you @justwagle and @scheherzade), the code ran, but I don't see the dot file (2018-10-02-tree.dot isn't in the directory). Any idea where it would be?

**Comment by hkaiser** [comment]
@kawilliams it should end up in the current working directory. But you can always specify a full file path as the target.

**Comment by kawilliams** [comment]
I tried specifying the full path (with and without quotes):
```
[kwilliams@rostam1 bin]$ ./physl als.physl \"MovieLens.csv\" 10 100 0.1 10 1 40 0 --dump-newick-tree=\"~/home/kwilliams/software/phylanx/build/bin/2018-10-02-tree.txt\"
```
No file is created. I can save the --performance data into a file, but then we're back to the human needing to cut-and-paste the tree data and performance data out from the file
```
[kwilliams@rostam1 bin]$ ./physl als.physl \"MovieLens.csv\" 10 100 0.1 10 1 40 0 --performance >> perf.txt
```
(perf.txt is created and contains the normal performance output)

**Comment by hkaiser** [comment]
@kawilliams please also specify the option `--performance`. Otherwise no output will be generated. I will add a corresponding message to the interpreter.

**Comment by hkaiser** [comment]
@kawilliams please see #611 for a fix.

**Comment by hkaiser** [comment]
This should have been addressed by merging #611
